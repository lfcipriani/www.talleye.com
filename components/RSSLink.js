import Link from 'next/link';

export default function RSSLink({ size }) {
  return (
    <>
      <Link href="/posts-rss.xml" locale="en">
        <a target="_blank" title="RSS feed">
          <span className="visually-hidden">RSS feed</span>
          <svg
            xmlns="http://www.w3.org/2000/svg"
            width={size || '35'}
            height={size || '35'}
            viewBox="0 0 256 256"
            className="svgIcon"
            aria-hidden="true"
            focusable="false"
          >
            <rect width="256" height="256" fill="none"></rect>
            <path
              d="M48,144a64,64,0,0,1,64,64"
              fill="none"
              stroke="#284B63"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="16"
              className="stroke"
            ></path>
            <path
              d="M48,96A112,112,0,0,1,160,208"
              fill="none"
              stroke="#284B63"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="16"
              className="stroke"
            ></path>
            <path
              d="M48,48A160,160,0,0,1,208,208"
              fill="none"
              stroke="#284B63"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="16"
              className="stroke"
            ></path>
            <circle cx="52" cy="204" r="12" className="fill"></circle>
          </svg>
        </a>
      </Link>
      <style jsx>{`
        .stroke {
          stroke: var(--color-icon);
        }
        .fill {
          fill: var(--color-icon);
        }
      `}</style>
    </>
  );
}
