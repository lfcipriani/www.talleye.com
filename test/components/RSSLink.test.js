import renderer from 'react-test-renderer';
import RSSLink from '../../components/RSSLink';

describe('RSSLink snapshot', () => {
  it('renders without size', () => {
    const component = renderer.create(<RSSLink />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with size', () => {
    const component = renderer.create(<RSSLink size="42" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
