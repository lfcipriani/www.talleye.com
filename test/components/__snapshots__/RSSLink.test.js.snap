// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RSSLink snapshot renders with size 1`] = `
<a
  className="jsx-3248159136"
  href="/posts-rss.xml"
  onClick={[Function]}
  onMouseEnter={[Function]}
  target="_blank"
  title="RSS feed"
>
  <span
    className="jsx-3248159136 visually-hidden"
  >
    RSS feed
  </span>
  <svg
    aria-hidden="true"
    className="jsx-3248159136 svgIcon"
    focusable="false"
    height="42"
    viewBox="0 0 256 256"
    width="42"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      className="jsx-3248159136"
      fill="none"
      height="256"
      width="256"
    />
    <path
      className="jsx-3248159136 stroke"
      d="M48,144a64,64,0,0,1,64,64"
      fill="none"
      stroke="#284B63"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="16"
    />
    <path
      className="jsx-3248159136 stroke"
      d="M48,96A112,112,0,0,1,160,208"
      fill="none"
      stroke="#284B63"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="16"
    />
    <path
      className="jsx-3248159136 stroke"
      d="M48,48A160,160,0,0,1,208,208"
      fill="none"
      stroke="#284B63"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="16"
    />
    <circle
      className="jsx-3248159136 fill"
      cx="52"
      cy="204"
      r="12"
    />
  </svg>
</a>
`;

exports[`RSSLink snapshot renders without size 1`] = `
<a
  className="jsx-3248159136"
  href="/posts-rss.xml"
  onClick={[Function]}
  onMouseEnter={[Function]}
  target="_blank"
  title="RSS feed"
>
  <span
    className="jsx-3248159136 visually-hidden"
  >
    RSS feed
  </span>
  <svg
    aria-hidden="true"
    className="jsx-3248159136 svgIcon"
    focusable="false"
    height="35"
    viewBox="0 0 256 256"
    width="35"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      className="jsx-3248159136"
      fill="none"
      height="256"
      width="256"
    />
    <path
      className="jsx-3248159136 stroke"
      d="M48,144a64,64,0,0,1,64,64"
      fill="none"
      stroke="#284B63"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="16"
    />
    <path
      className="jsx-3248159136 stroke"
      d="M48,96A112,112,0,0,1,160,208"
      fill="none"
      stroke="#284B63"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="16"
    />
    <path
      className="jsx-3248159136 stroke"
      d="M48,48A160,160,0,0,1,208,208"
      fill="none"
      stroke="#284B63"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="16"
    />
    <circle
      className="jsx-3248159136 fill"
      cx="52"
      cy="204"
      r="12"
    />
  </svg>
</a>
`;
